src/components/player/progress.ts

```diff
@@ -1,96 +1,68 @@
 import Gtk from "gi://Gtk?version=4.0";
 import GObject from "gi://GObject";
-import Adw from "gi://Adw";
+
+import { SignalListeners } from "src/util/signal-listener";
+import { get_player } from "src/application";
 
 export class PlayerProgressBar extends Gtk.ProgressBar {
   static {
     GObject.registerClass({
       GTypeName: "PlayerProgressBar",
-      Properties: {
-        duration: GObject.ParamSpec.double(
-          "duration",
-          "Duration",
-          "Duration in nanoseconds",
-          GObject.ParamFlags.READABLE,
-          0,
-          Number.MAX_SAFE_INTEGER,
-          0,
-        ),
-        value: GObject.ParamSpec.double(
-          "value",
-          "Value",
-          "Value in nanoseconds",
-          GObject.ParamFlags.READWRITE,
-          0,
-          Number.MAX_SAFE_INTEGER,
-          0,
-        ),
-        buffering: GObject.ParamSpec.boolean(
-          "buffering",
-          "Buffering",
-          "Whether the player is buffering",
-          GObject.ParamFlags.READWRITE,
-          false,
-        ),
-        animation: GObject.ParamSpec.object(
-          "animation",
-          "Animation",
-          "The animation object",
-          GObject.ParamFlags.READWRITE,
-          Adw.TimedAnimation,
-        ),
-      },
-      Signals: {},
     }, this);
   }
 
   constructor() {
     super({
-      hexpand: true,
       valign: Gtk.Align.START,
     });
 
     this.add_css_class("osd");
   }
 
-  get value() {
-    return this.fraction * this.duration;
-  }
+  private update_fraction() {
+    const player = get_player();
 
-  set value(value: number) {
-    this.fraction = Math.max(Math.min(value / this.duration, 1), 0) || 0;
-  }
+    this.fraction = Math.max(
+      Math.min(
+        player.initial_seek_to ?? player.timestamp / player.duration,
```

Comment:
```suggestion
        (player.initial_seek_to ?? player.timestamp) / player.duration,
```
![image](https://github.com/vixalien/muzika/assets/8540562/7158bd61-2a2c-4f41-ae08-be5886edf180) :sweat_smile: